core/rpc/rpc.go

```diff
@@ -138,11 +140,20 @@ func (c *Client) CallRaw(ctx context.Context, path string, request interface{})
 	}
 
 	if resp.StatusCode < 200 || resp.StatusCode >= 300 {
-		resp.Body.Close()
-		return nil, ErrStatusCode{
+		resErr := ErrStatusCode{
 			URL:        cleanedURLString(u),
 			StatusCode: resp.StatusCode,
 		}
+
+		// Attach formatted error message, if available
+		var errData httperror.Response
+		err := json.NewDecoder(resp.Body).Decode(&errData)
+		if err == nil && errData.ChainCode != "" {
+			resErr.ErrorData = &errData
+		}
+		resp.Body.Close()
```

Comment:
This is unlikely to matter but can you put this call back at the top of the if body in a defer? That way an inadvertent panic in between  there and here won't leak a connection.